processing/src/test/java/org/apache/druid/segment/join/HashJoinSegmentStorageAdapterTest.java

```diff
@@ -1288,103 +1313,1163 @@ public void test_makeCursors_errorOnNonKeyBasedJoin()
     );
   }
 
-  private JoinableClause factToCountryNameUsingIsoCodeLookup(final JoinType joinType)
+  // Filter push down tests
```

Comment:
The test coverage of the filter push down code is great!

What do you think about making many of these test cases as unit tests in `JoinFilterAnalyzerTest` instead? That way it's more straightforward to map the test case to the relevant code.

Reply:
I split out the segment setup and some helper methods out of `HashJoinSegmentStorageAdapterTest` into `BaseHashJoinSegmentStorageAdapterTest` since the JoinFilterAnalyzer needs a HashJoinSegmentStorageAdapter, both `JoinFilterAnalyzerTest` and `HashJoinSegmentStorageAdapterTest` now extend the base.

The filter push down tests have been moved into `JoinFilterAnalyzerTest`